include/crow/json.h

```diff
@@ -1851,9 +1851,9 @@ namespace crow
                             } f_state;
                             char outbuf[128];
 #ifdef _MSC_VER
-                            sprintf_s(outbuf, sizeof(outbuf), "%f", v.num.d);
+                            sprintf_s(outbuf, sizeof(outbuf), "%.*g", DBL_DECIMAL_DIG, v.num.d);
 #else
-                            snprintf(outbuf, sizeof(outbuf), "%f", v.num.d);
+                            snprintf(outbuf, sizeof(outbuf), "%.*g", DECIMAL_DIG, v.num.d);
```

Comment:
shouldn't DBL_DECIMAL_DIG also used here?

Reply:
Interesting.
`DECIMAL_DIG` is under C++11 standard, while `DBL_DECIMAL_DIG` is C++17 and it doesn't work in my building environment. Let me make the change and figure out my issue later on.